### PR TITLE
ci: disable git tag and push for alpha release

### PR DESCRIPTION
**Pull request by lihbr**
## Context

Right now, when publishing an alpha through the CI, lots of things happen. Among them:
- A git tag is created and pushed (e.g. https://github.com/prismicio/slice-machine/releases/tag/start-slicemachine%400.6.1-dev-plugins-m2-validation.0)
- A commit is made (e.g. https://github.com/prismicio/slice-machine/commit/acfe527c091dbc11c5e6a1539042b28a9cd36f31)

Those create some issues:
- The git repository is "spammed" with short-living version tags, making navigating git tags complex
- The git repository is "spammed" with release commits, sometimes corrupting `package-lock.json` files

## The Solution

Lerna allows us to disable the git tag and commit behavior using flags: https://github.com/lerna/lerna/tree/f6e7a13e60fefc523d701efddfcf0ed41a77749b/commands/version

By using them we're essentially creating "virtual" alpha releases from GitHub point of view, not polluting GitHub with them (only NPM)

However, this creates collision issues (thanks @bapmrl!) because Lerna would not be able to determine the prerelease number since we stop committing them.

Therefore we can instead fallback on the "canary" flag of the publish command, which does just that and is tailored for per-commit release: https://github.com/lerna/lerna/blob/f6e7a13e60fefc523d701efddfcf0ed41a77749b/commands/publish/README.md#--canary, this is what this PR implements.

## Checklist before requesting a review
- [ ] I hereby declare my code ready for review.
- [ ] If it is a critical feature, I have added tests.
- [ ] The CI is successful.
- [ ] If there could backward compatibility issues, it has been discussed and planned.

### CODE REVIEW

**Review by bapmrl — APPROVED** [review]
Thank you! However, I don't know Lerna and wonder how it knows which versions already exist? Does it query `npm` before bumping?

**Review by bapmrl — APPROVED** [review]

**Comment by lihbr** [comment]
Alright, I'm almost there but I think it's related to Lerna version where the canary flag tries to publish versions that can't be published (`1.0.0-alpha.0+123456` is sanitized to `1.0.0-alpha.0` which is not what we want):

---

image

---

image

---

Related: https://github.com/lerna/lerna/issues/2060#issuecomment-569968758, https://github.com/lerna/lerna/issues/3554

tl;dr; Lerna canary appears to have some issue, even in recent versions of Lerna

**Comment by lihbr** [comment]
OK, I got it to work using this fix: https://github.com/lerna/lerna/issues/2060#issuecomment-658182391, see this job's result: https://github.com/prismicio/slice-machine/actions/runs/4566429321/jobs/8058917521

However, this still references a corrupted package version in package jsons, see the code here for example: https://www.npmjs.com/package/start-slicemachine/v/0.6.1-dev-ci.00aa6f7.0?activeTab=code

So I'm out of ideas on how to run that now :(

**Comment by lihbr** [comment]
Closing until we get a better idea o/
